### DISCUSSION TITLE
How the versioning works?

### DISCUSSION THREAD

**Discussion by jcornaz** [boozook/playdate#93] [Q&A] [answered]
Hi, 

I'm a bit confused because the [github releases](https://github.com/boozook/playdate/releases) (e.g. `2023.09.20`) don't match the ones on [crates.io](https://crates.io/crates/playdate-sys/versions) (e.g. `0.1.9`).

How does that work?

**Comment by boozook** [comment, marked as answer]
Hi 👋 

Versioning is in cargo packages and crates-io. GitHub-releases is just kind of milestones, but tags in date format to not break semver compatibility ;)

I made this decision because it's a monorepo without a single main package to define the version for all repo, as well as tag.

Yes, of course I could put tags in "crate-name-version" format every time I update any box, but it could be messy.

If you have an idea on how to do it better, I'd be happy to.

**Reply by boozook** [reply]
In addition, I wouldn't need GitHub releases at all if it weren't for the handy feature of automatically generating changelogs including mentions of contributors. It's convenient and easy and that's why I turned on the releases.

**Reply by jcornaz** [reply]
Ok. Thanks for the clarification.

I understand the complexity of versioning with a mono-repo approach. If you're interested, [cargo smart-release](https://github.com/Byron/cargo-smart-release) may help, as I understand it was specifically designed to solve [this issue](https://github.com/Byron/cargo-smart-release#made-for-this-workflow).

**Reply by boozook** [reply]
Thanks for your recommendation, really! 
I probably will 🤔

Currently, I'm already using [another thing][] over there and it does what I need to do in [six lines][].

[six lines]: https://github.com/boozook/playdate/blob/main/.github/workflows/publish.yml#L52-L57
[another thing]: http://github.com/katyo/publish-crates

**Comment by boozook** [comment]
@jcornaz,
Just have to notice that now the releases contain builds.
